resources/assets/components/Post/index.js

```diff
@@ -20,12 +20,18 @@ class Post extends React.Component {
 
     this.state = {
       loading: false,
-      post: this.props.post
+      post: this.props.post,
     };
 
     this.api = new RestApiClient;
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.post !== this.state.post) {
+      this.setState({ post: nextProps.post });
```

Comment:
This makes me think that `post` should just be a prop, and not duplicated in the component's state... instead, let the parent handle making the API request to rotate it and send back down the updated data via props.

Reply:
yea i think that makes more sense. This definitely should be refactored. Will tackle that this week since I have some bandwidth this sprint!